src/adapters/stickyadstv.js

```diff
@@ -0,0 +1,265 @@
+  var SCREENROLL_URL = "//cdn.stickyadstv.com/prime-time/screen-roll.min.js";
+  
+  window.stickyadstv_cache = {};
+
+  function _callBids(params) {
+     
+    var bids = params.bids || [];
+    for (var i = 0; i < bids.length; i++) {
+      var bid = bids[i];
+      // Send out bid request for each bid given its tag IDs and query strings
+
+      if(bid.placementCode && bid.params.zoneId) {
+        sendBidRequest(bid);
+      }
+      else {
+        console.warn("StickyAdsTV: Missing mandatory field(s).");
+      }
+
+    }
+  }
+
+  function sendBidRequest(bid){
+
+    var placementCode = bid.placementCode;
+
+    var integrationType = bid.params.format ? bid.params.format : "inbanner"; 
+    var urltoLoad = MUSTANG_URL;
+
+    if(integrationType === "intext-roll"){
+      urltoLoad = INTEXTROLL_URL;
+    }
+    if(integrationType === "screen-roll"){
+      urltoLoad = SCREENROLL_URL;
+    }
+
+    var bidRegistered = false;
+    adloader.loadScript(urltoLoad, function(){
+
+      getBid(bid, function(bidObject){
+
+        if(!bidRegistered){
+          bidRegistered = true;
+          bidmanager.addBidResponse(placementCode, bidObject);
+        }
+        
+      });
+    }, true);
+  }
+
+  function getBid(bid, callback){
+    var zoneId = bid.params.zoneId || bid.params.zone; //accept both
+    var size = getBiggerSize(bid.sizes);
+
+    var vastLoader = new window.com.stickyadstv.vast.VastLoader();
+    bid.vast = window.stickyadstv_cache[bid.placementCode] = vastLoader.getVast();
+
+    var vastCallback = {
+      onSuccess : bind(function(){
+        //'this' is the bid here
+        var bid = this;
+        
+        var adHtml = formatAdHTML(bid,size);
+        var price = extractPrice(bid.vast);
+
+        callback(formatBidObject(true, price, adHtml, size[0], size[1]));
+        
+      },bid),
+      onError : bind(function(){
+        callback(formatBidObject(false));
+      },bid)
+    };
+
+    var config = {
+      zoneId:zoneId,
+      playerSize:size[0]+"x"+size[1],
+      vastUrlParams: bid.params.vastUrlParams,
+      componentId: "prebid-sticky"+(bid.params.format ? "-"+bid.params.format : "")
+    };
+
+    if(bid.params.format === "screen-roll"){
+      //in screenroll case we never use the original div size.
+      config.playerSize = window.com.stickyadstv.screenroll.getPlayerSize();
+    }
+    
+    vastLoader.load(config, vastCallback);
+  }
+
+  function getBiggerSize(array){
+    var result = [1,1];
+    for(var i = 0; i< array.length; i++){
+      if(array[i][0]*array[i][1] > result[0]*result[1]){
+        result = array[i];
+      }
+    }
+    return result;
+  }
+  
+  var formatInBannerHTML = function(bid,size){
+    var placementCode = bid.placementCode;
+    
+    var divHtml = "<div id=\"stickyadstv_prebid_target\"></div>";
+
+    var script = "<script type='text/javascript'>"+
+       
+    "var vast =  window.top.stickyadstv_cache[\""+placementCode+"\"];"+
```

Comment:
Use of `window.top` will not work when running in a cross domain iframe, which is increasingly more common. Delivery of creatives into a SafeFrame, for example, or when integrating Prebid with an AMP Page.